[RLlib] TestCheckBaseEnv.test_check_env_step_incorrect_error doesn't work on Windows

### What happened + What you expected to happen

```
env = <ray.rllib.env.multi_agent_env.MultiAgentEnvWrapper object at 0x0000028114F50DC0>

    @DeveloperAPI
    def check_env(env: EnvType) -> None:
        """Run pre-checks on env that uncover common errors in environments.

        Args:
            env: Environment to be checked.

        Raises:
            ValueError: If env is not an instance of SUPPORTED_ENVIRONMENT_TYPES.
            ValueError: See check_gym_env docstring for details.
        """
        from ray.rllib.env import (
            BaseEnv,
            MultiAgentEnv,
            RemoteBaseEnv,
            VectorEnv,
            ExternalMultiAgentEnv,
            ExternalEnv,
        )

        if hasattr(env, "_skip_env_checking") and env._skip_env_checking:
            # This is a work around for some environments that we already have in RLlb
            # that we want to skip checking for now until we have the time to fix them.
            if log_once("skip_env_checking"):
                logger.warning("Skipping env checking for this experiment")
            return

        try:
            if not isinstance(
                env,
                (
                    BaseEnv,
                    gym.Env,
                    MultiAgentEnv,
                    RemoteBaseEnv,
                    VectorEnv,
                    ExternalMultiAgentEnv,
                    ExternalEnv,
                    ActorHandle,
                ),
            ):
                raise ValueError(
                    "Env must be of one of the following supported types: BaseEnv, "
                    "gym.Env, "
                    "MultiAgentEnv, VectorEnv, RemoteBaseEnv, ExternalMultiAgentEnv, "
                    f"ExternalEnv, but instead is of type {type(env)}."
                )
            if isinstance(env, MultiAgentEnv):
                check_multiagent_environments(env)
            elif isinstance(env, gym.Env):
                check_gym_environments(env)
            elif isinstance(env, BaseEnv):
                check_base_env(env)
            else:
                logger.warning(
                    "Env checking isn't implemented for VectorEnvs, RemoteBaseEnvs, "
                    "ExternalMultiAgentEnv, ExternalEnvs or environments that are "
                    "Ray actors."
                )
        except Exception:
            actual_error = traceback.format_exc()
>           raise ValueError(
                f"{actual_error}\n"
                "The above error has been found in your environment! "
                "We've added a module for checking your custom environments. It "
                "may cause your experiment to fail if your environment is not set up"
                "correctly. You can disable this behavior by setting "
                "`disable_env_checking=True` in your environment config "
                "dictionary. You can run the environment checking module "
                "standalone by calling ray.rllib.utils.check_env([env])."
            )
E           ValueError: Traceback (most recent call last):
E             File "c:\programdata\anaconda3\envs\pytorch\lib\site-packages\ray\rllib\utils\pre_checks\env.py", line 73, in check_env
E               check_base_env(env)
E             File "c:\programdata\anaconda3\envs\pytorch\lib\site-packages\ray\rllib\utils\pre_checks\env.py", line 349, in check_base_env
E               _check_if_multi_env_dict(env, next_obs, "step, next_obs")
E             File "c:\programdata\anaconda3\envs\pytorch\lib\site-packages\ray\rllib\utils\pre_checks\env.py", line 480, in _check_if_multi_env_dict
E               _check_if_element_multi_agent_dict(
E             File "c:\programdata\anaconda3\envs\pytorch\lib\site-packages\ray\rllib\utils\pre_checks\env.py", line 499, in _check_if_element_multi_agent_dict
E               raise ValueError(error)
E           ValueError: The element returned by step, next_obs contains values that are not MultiAgentDicts. Instead, they are of type: <class 'int'>
E
E           The above error has been found in your environment! We've added a module for checking your custom environments. It may cause your experiment to fail if your environment is not set upcorrectly. You can disable this behavior by setting `disable_env_checking=True` in your environment config dictionary. You can run the environment checking module standalone by calling ray.rllib.utils.check_env([env]).

c:\programdata\anaconda3\envs\pytorch\lib\site-packages\ray\rllib\utils\pre_checks\env.py:82: ValueError
```

### Versions / Dependencies

python 3.8.1
tensorflow 2.5.0
Ray master HEAD

### Reproduction script

pytest rllib/utils/tests/test_check_env.py

### Issue Severity

_No response_

--- a/rllib/BUILD
+++ b/rllib/BUILD
@@ -1807,6 +1807,20 @@ py_test(
     srcs = ["utils/replay_buffers/tests/test_segment_tree_replay_buffer_api.py"]
 )
 
+py_test(
+    name = "test_check_env",
+    tags = ["team:rllib", "utils"],
+    size = "small",
+    srcs = ["utils/tests/test_check_env.py"]
+)
+
+py_test(
+    name = "test_check_multi_agent",
+    tags = ["team:rllib", "utils"],
+    size = "small",
+    srcs = ["utils/tests/test_check_multi_agent.py"]
+)
+
 # --------------------------------------------------------------------
 # rllib/tests/ directory
 #
@@ -3285,8 +3299,6 @@ py_test_module_list(
     "examples/env/tests/test_matrix_sequential_social_dilemma.py",
     "examples/env/tests/test_wrappers.py",
     "execution/tests/test_mixin_multi_agent_replay_buffer.py",
-    "utils/tests/test_check_env.py",
-    "utils/tests/test_check_multi_agent.py",
     "utils/tests/test_errors.py",
     "utils/tests/test_utils.py",
   ],

--- a/rllib/utils/tests/test_check_env.py
+++ b/rllib/utils/tests/test_check_env.py
@@ -292,7 +292,7 @@ class TestCheckBaseEnv:
         with pytest.raises(
             ValueError,
             match="The element returned by step, "
-            "next_obs has values that are not"
+            "next_obs contains values that are not"
             " MultiAgentDicts",
         ):
             check_env(env)

--- a/rllib/utils/tests/test_check_multi_agent.py
+++ b/rllib/utils/tests/test_check_multi_agent.py
@@ -42,7 +42,7 @@ class TestCheckMultiAgent(unittest.TestCase):
         config = {"multiagent": {"replay_mode": "invalid_value"}}
         self.assertRaisesRegex(
             ValueError,
-            "config.multiagent.replay_mode must be",
+            "`config.multiagent.replay_mode` must be",
             lambda: PG(config, env="CartPole-v0"),
         )
 

